docs/developers/guides/token.md

```diff
@@ -0,0 +1,1287 @@
+    const tkn = new Token();
+
+    // set contract_call authority for CONTRACT_ID to true so that we can mint tokens
+    const authContractId = new MockVM.MockAuthority(
+      authority.authorization_type.contract_call,
+      CONTRACT_ID,
+      true
+    );
+
+    // set contract_call authority for MOCK_ACCT1 to true so that we can transfer tokens
+    const authMockAcct1 = new MockVM.MockAuthority(
+      authority.authorization_type.contract_call,
+      MOCK_ACCT1,
+      true
+    );
+    MockVM.setAuthorities([authContractId, authMockAcct1]);
+
+    // mint tokens
+    const mintArgs = new token.mint_arguments(MOCK_ACCT1, 123);
+    tkn.mint(mintArgs);
+
+    // save the MockVM state because the transfer is going to revert the transaction
+    MockVM.commitTransaction();
+
+    expect(() => {
+      // try to transfer tokens
+      const tkn = new Token();
+      const transferArgs = new token.transfer_arguments(
+        MOCK_ACCT1,
+        MOCK_ACCT1,
+        10
+      );
+      tkn.transfer(transferArgs);
+    }).toThrow();
+
+    expect(MockVM.getErrorMessage()).toStrictEqual("Cannot transfer to self");
+
+    // check balances
+    let balanceArgs = new token.balance_of_arguments(MOCK_ACCT1);
+    let balanceRes = tkn.balance_of(balanceArgs);
+    expect(balanceRes.value).toBe(123);
+  });
+
+  it("should not transfer if insufficient balance", () => {
+    const tkn = new Token();
+
+    // set contract_call authority for CONTRACT_ID to true so that we can mint tokens
+    const authContractId = new MockVM.MockAuthority(
+      authority.authorization_type.contract_call,
+      CONTRACT_ID,
+      true
+    );
+
+    // set contract_call authority for MOCK_ACCT1 to true so that we can transfer tokens
+    const authMockAcct1 = new MockVM.MockAuthority(
+      authority.authorization_type.contract_call,
+      MOCK_ACCT1,
+      true
+    );
+    MockVM.setAuthorities([authContractId, authMockAcct1]);
+
+    // mint tokens
+    const mintArgs = new token.mint_arguments(MOCK_ACCT1, 123);
+    tkn.mint(mintArgs);
+
+    // save the MockVM state because the transfer is going to revert the transaction
+    MockVM.commitTransaction();
+
+    expect(() => {
+      // try to transfer tokens
+      const tkn = new Token();
+      const transferArgs = new token.transfer_arguments(
+        MOCK_ACCT1,
+        MOCK_ACCT2,
+        456
+      );
+      tkn.transfer(transferArgs);
+    }).toThrow();
+
+    expect(MockVM.getErrorMessage()).toStrictEqual(
+      "'from' has insufficient balance"
+    );
+
+    // check balances
+    let balanceArgs = new token.balance_of_arguments(MOCK_ACCT1);
+    let balanceRes = tkn.balance_of(balanceArgs);
+    expect(balanceRes.value).toBe(123);
+
+    balanceArgs = new token.balance_of_arguments(MOCK_ACCT2);
+    balanceRes = tkn.balance_of(balanceArgs);
+    expect(balanceRes.value).toBe(0);
+  });
+});
+
+```
+
+Build and debug the project using the following command:
+
+```sh
+yarn build:debug
```

Comment:
The calculator and immutable guides build using this command: `koinos-sdk-as-cli build debug 0`. Let's use it in case the sdk-as-cli changes backend implementations the guide will still work.